Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM neurodebian:stretch
+
+ARG DEBIAN_FRONTEND="noninteractive"
+
+RUN apt-get update -qq && apt-get install -y --no-install-recommends less \
+                                                                     ca-certificates \
+                                                                     wget \
+                                                                     unzip \
+                                                                     octave \
+                                                                     octave-general \
+                                                                     octave-signal \
+                                                                     octave-image \
+                                                                     liboctave-dev \
+                       && apt-get clean \
+                       && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# download HCP workbench (1.3.0) for CIFTI support
+WORKDIR /opt
+RUN wget -q https://ftp.humanconnectome.org/workbench/workbench-linux64-v1.3.0.zip -O wb.zip \
+    && unzip wb.zip \
+    && rm wb.zip
+
+# copy repo into /opt/palm
+WORKDIR palm
+COPY . .
+
+# recompile mex files to be compatible with our system
+WORKDIR fileio
+RUN cd \@gifti/private \
+    && mkoctfile --mex zstream.c \ 
+    && mkoctfile --mex miniz.c \
```

Comment:
mini.z is a single-file library that is included in zstream.c so it doesn't have to be compiled separately (and it does not contain a MEX interface).